test/dynamo/test_sources.py

```diff
@@ -22,6 +22,29 @@ def test_is_local(self):
         self.assertTrue(is_from_local_source(attr_x_a))
         self.assertEqual(is_from_local_source(attr_y_b), False)
 
+    def test_property_closure(self):
+        def external_property():
+            closed_value = 7
+
+            def internal_function(self):
+                return closed_value
+
+            return internal_function
+
+        class Elements:
+            myprop = property(external_property())
+
+        def func(elements):
+            if not elements.myprop:
+                return torch.tensor([1, 2, 3])
+            else:
+                return torch.tensor([4, 5, 6])
+
+        e = Elements()
+        a = func(e)
+        b = torch.compile(func)(e)
```

Comment:
test/dynamo tests either use torch.compile(backend=eager) or torch._dynamo.optimize. using the full torch.compile is slow.

Reply:
Also, can we put fullgraph=True on this?

Reply:
https://github.com/pytorch/pytorch/pull/119921